libs/ledger-live-common/src/api/Ethereum.ts

```diff
@@ -158,84 +166,100 @@ export const apiForCurrency = (currency: CryptoCurrency): API => {
   }
 
   return {
-    async getTransactions(address, block_hash, batch_size = 2000) {
-      let { data } = await network({
-        method: "GET",
-        url: URL.format({
-          pathname: `${baseURL}/addresses/${address}/transactions`,
-          query: {
-            batch_size,
-            noinput: true,
-            no_input: true,
-            no_token: true,
-            filtering: true,
-            block_hash,
-          },
+    async getTransactions(
+      address,
+      block_hash,
+      batch_size = 2000
+    ): Promise<Tx[]> {
+      const txs = Promise.all([
+        network({
+          method: "GET",
+          url: URL.format({
+            pathname: `${baseURL}/address/${address}/txs/pending`,
+            query: {
+              batch_size,
+              noinput: true,
+              no_input: true,
+              no_token: true,
+              filtering: true,
+              block_hash,
+            },
+          }),
         }),
-        transformResponse: JSONBigNumber.parse,
-      });
-
-      // v3 have a bug that still includes the tx of the paginated block_hash, we're cleaning it up
-      if (block_hash) {
-        data = {
-          ...data,
-          txs: data.txs.filter(
-            (tx) => !tx.block || tx.block.hash !== block_hash
-          ),
-        };
-      }
-
-      return data;
+        network({
+          method: "GET",
+          url: URL.format({
+            pathname: `${baseURL}/address/${address}/txs`,
+            query: {
+              batch_size,
+              noinput: true,
+              no_input: true,
+              no_token: true,
+              filtering: true,
+              block_hash,
+            },
+          }),
+        }),
+      ]);
+      const [pendingTxData, txData] = await txs;
+      return pendingTxData.data.concat(txData.data);
```

Comment:
[https://stackoverflow.com/questions/5080028/what-is-the-most-efficient-way-to-concatenate-n-arrays](https://stackoverflow.com/questions/5080028/what-is-the-most-efficient-way-to-concatenate-n-arrays)[
https://thisthat.dev/concat-vs-push/#:~:text=concat%20returns%20a%20new%20array,difference%20in%20term%20of%20performance.](
https://thisthat.dev/concat-vs-push/#:~:text=concat%20returns%20a%20new%20array,difference%20in%20term%20of%20performance.)

Array.prototype.push.apply(firstArray, secondArray) is limited to 65k elements in the second array (pending txs here) if you think this is troublesome, i will switch to the lower performance alternative a.push(...b)

Reply:
Completely unlikely to have more than 65k pending txs returned by backend, so this should be fine.
I'm not sure either it improves anything :sweat_smile: it was more in case it help solve any perf issue.

Reply:
Should be 100x times more performance according to jsperf, all links lead to a 500 error atm tho